examples/counter/crate/src/lib.rs

```diff
@@ -28,59 +28,53 @@ impl Render for Counter {
     {
         let val = bumpalo::format!(in bump, "{}", self.val);
 
-        let increment = Node::element(
-            bump,
-            "button",
-            [Attribute {
-                name: "data-action",
-                value: "increment",
-            }],
-            [Node::text("+")],
-        );
-
-        let decrement = Node::element(
-            bump,
-            "button",
-            [Attribute {
-                name: "data-action",
-                value: "decrement",
-            }],
-            [Node::text("-")],
-        );
-
         Node::element(
             bump,
             "div",
             [],
-            [increment, Node::text(val.into_bump_str()), decrement],
+            [],
+            [
+                Node::element(
+                    bump,
+                    "button",
+                    [on(bump, "click", |root, vdom, _event| {
+                        root.unwrap_mut::<Counter>().increment();
+                        vdom.schedule_render();
+                    })],
+                    [],
+                    [Node::text("+")],
+                ),
+                Node::text(val.into_bump_str()),
+                Node::element(
+                    bump,
+                    "button",
+                    [on(bump, "click", |root, vdom, _event| {
+                        root.unwrap_mut::<Counter>().decrement();
+                        vdom.schedule_render();
```

Comment:
Also to confirm from before, what's the use case for *not* scheduling a rerender on a handled event?

Reply:
If the event only sends a request to the server and doesn't change the UI, or you detect that a re-render is unnecessary for whatever other reason.

It is the most general thing, since we don't expose any "unschedule render" methods (and I don't think that would make sense).